services/horizon/internal/db2/history/offers.go

```diff
@@ -83,28 +86,45 @@ func (q *Q) GetOffers(ctx context.Context, query OffersQuery) ([]Offer, error) {
 
 // StreamAllOffers loads all non deleted offers
 func (q *Q) StreamAllOffers(ctx context.Context, callback func(Offer) error) error {
+	lastID := int64(math.MinInt64)
```

Comment:
Zero seems to be good given offer IDs start at 1?
```suggestion
	lastID := int64(0)
```